crates/interpreter/src/host/dummy.rs

```diff
@@ -64,10 +67,23 @@ impl Host for DummyHost {
     }
 
     #[inline]
+    #[cfg(not(feature = "scroll"))]
     fn code_hash(&mut self, __address: Address) -> Option<(B256, bool)> {
         Some((KECCAK_EMPTY, false))
     }
 
+    #[inline]
+    #[cfg(feature = "scroll")]
+    fn code_hash(&mut self, __address: Address) -> Option<(B256, bool)> {
+        Some((POSEIDON_EMPTY, false))
+    }
+
+    #[inline]
+    #[cfg(feature = "scroll")]
+    fn keccak_code_hash(&mut self, __address: Address) -> Option<(B256, bool)> {
+        Some((KECCAK_EMPTY, false))
+    }
+
```

Comment:
Using both feature and not feature seems like an antipattern. Will be a drag in long term maintainance.
```
#[inline]
    fn code_hash(&mut self, __address: Address) -> Option<(B256, bool)> {
        #[cfg(feature = "scroll")]
        Some((POSEIDON_EMPTY, false))

        Some((KECCAK_EMPTY, false))
    }
    
    

```

this will also remove un- needed tags on top of code that uses code_hash function also